### PR TITLE
#161967133 users should be able to like and or dislike articles

### PR DESCRIPTION
**Pull request by jillo-abdullahi**
**What does this PR do?**

Enable users to post their opinions by liking or disliking an article.

**Description of tasks to be completed**

1. Implement feature for authenticated users to like or dislike articles.

2. Similar opinion posted twice will remove the like/dislike.

3. Write tests for this functionality.

**How should this be manually tested?**

Clone the repo from your terminal: `git clone https://github.com/andela/ah-code-titans.git`

Create a virtual environment:`python3 -m venv <name_of_environment>`

Activate the virtual environment:  `source <name_of_environment>/bin/activate`

Install project dependencies: `pip install -r requirements.txt`

Add the following environment variables and populate them with your settings:

- `export DB_NAME='choose_a_dbname'`
- `export DB_USER='db_username'`
- `export DB_PASSWORD='db_role_password'`
- `export DB_HOST='localhost or preffered host'`
- `export DB_PORT='server_port_number'`
- `export SECRET_KEY='your_secret_key'`
- `export EMAIL_HOST_USER="codetitans32@gmail.com"`
- `export EMAIL_HOST_PASSWORD='Codetitans@32'`

Switch to the appropriate branch: `git checkout ft-likedislike-article-161967133`
Run migrations: 

1. `python manage.py makemigrations`
2. `python manage.py migrate`

Run tests: `python manage.py test`

**What are the relevant Pivotal Tracker stories?**
[#161967133](https://www.pivotaltracker.com/story/show/161967133)

**Checklist**

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation

- [x] My changes generate no new warnings

**Screenshots**

screenshot 2018-12-13 at 16 53 48

screenshot 2018-12-13 at 16 53 31

### CODE REVIEW

File: authors/apps/likedislike/models.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    VOTES = (
+        (LIKE, 'like'),
+        (DISLIKE, 'dislike')
+    )
+
+    vote = models.SmallIntegerField(verbose_name="likes", choices=VOTES)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
```
Comment (bettblake08) [inline authors/apps/likedislike/models.py:53]: Can you please enlighten me on why we need the content_type, object_id and content_object?

**Review by bettblake08 — COMMENTED** [review]
I need some clarification for some of the code you have written.

File: authors/apps/likedislike/views.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        # Let's return some confirmation data
+        # along with a response status
+        return Response({
+            "article_slug": slug,
+            "article_id": obj.id,
+            "total_likes": obj.votes.likes().count(),
+            "total_dislikes": obj.votes.dislikes().count(),
+            "total_votes": obj.votes.sum_rating(),
+        },
```
Comment (denomash) [inline authors/apps/likedislike/views.py:71]: what is the essence of the `total_votes`

**Review by denomash — CHANGES_REQUESTED** [review]

File: authors/apps/likedislike/models.py (hunk 1 above)
Comment (bettblake08) [reply authors/apps/likedislike/models.py:53]: Add documentation for this three class attributes.

**Review by bettblake08 — CHANGES_REQUESTED** [review]

File: authors/apps/likedislike/views.py (hunk 2 above)
Comment (jillo-abdullahi) [reply authors/apps/likedislike/views.py:71]: @denomash It will help with getting the total likes/dislikes the article has received.

Comment (denomash) [reply authors/apps/likedislike/views.py:71]: ok

**Review by denomash — APPROVED** [review]

**Review by davishooly — APPROVED** [review]

**Comment by jillo-abdullahi** [comment]
@bettblake08 I have added the documentation that you have requested.

**Pull request merged**